include/boost/url/experimental/detail/router.hpp

```diff
@@ -0,0 +1,1230 @@
+        {
+            point = std::use_facet<
+                        std::numpunct<char>>(std::locale()).decimal_point();
+        }
+
+        switch (type_)
+        {
+            case 'd':
+                return is_token(dseg, "0123456789");
+            case 'x':
+                return is_token(dseg, "0123456789abcdef");
+            case 'X':
+                return is_token(dseg, "0123456789ABCDEF");
+            case 'o':
+                return is_token(dseg, "01234567");
+            case 'g':
+            case 'G':
+            case 'f':
+            case 'F':
+            case 'e':
+            case 'E':
+                return is_float_token(dseg, "0123456789", point, alternate_);
+            case 'a':
+                return is_float_token(dseg, "0123456789abcdef", point, alternate_);
+            case 'A':
+                return is_float_token(dseg, "0123456789ABCDEF", point, alternate_);
+            case 'b':
+            case 'B':
+                return is_token(dseg, "01");
+            case 'c':
+                return dseg.size() == 1;
+            case 'p':
+                return is_token(dseg, "0123456789abcdef");
+        }
+        return true;
+    }
+
+    bool
+    match_range(decode_view dseg) const
+    {
+        if (!dseg.starts_with("["))
+            return false;
+        dseg.remove_prefix(1);
+        if (!dseg.ends_with("]"))
+            return false;
+        dseg.remove_suffix(1);
+        while (!dseg.empty())
+        {
+            auto it0 = dseg.begin();
+            auto it = dseg.begin();
+            std::size_t dn = 0;
+            while (
+                it != dseg.end() &&
+                *it != ',')
+            {
+                ++it;
+                ++dn;
+            }
+            if (seg_type_ == seg_type::builtin_range)
+            {
+                if (!match_builtin(
+                        *make_pct_string_view_unsafe(
+                            it0.base(),
+                            it.base() - it0.base(),
+                            dn)))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                if (!match_chrono(
+                        *make_pct_string_view_unsafe(
+                            it0.base(),
+                            it.base() - it0.base(),
+                            dn)))
+                {
+                    return false;
+                }
+            }
+            dseg = *make_pct_string_view_unsafe(
+                it.base(),
+                dseg.end().base() - it.base(),
+                dseg.size() - dn);
+            if (dseg.empty())
+                break;
+            if (!dseg.starts_with(','))
+                return false;
+            dseg.remove_prefix(1);
+            if (dseg.starts_with(" "))
+                dseg.remove_prefix(1);
+        }
+        return true;
+    }
+
+    bool
+    match_chrono(decode_view dseg) const
```

Comment:
wtf

Reply:
I implemented the fmtlib syntax in https://fmt.dev/latest/syntax.html.

A few types didn't look so useful but leaving it incomplete just to remove a few types seemed confusing to the user.

Reply:
Nope there was no need to duplicate fmtlib and even if there was, we ship with strings/unsigned and then only add additional things on a case by case basis with guidance from stakeholders.